deepchecks/checks/performance/simple_model_comparison.py

```diff
@@ -126,27 +131,30 @@ def _find_score(self, train_ds: Dataset, test_ds: Dataset, task_type: ModelType,
             elif task_type in (ModelType.BINARY, ModelType.MULTICLASS):
                 clf = DecisionTreeClassifier(max_depth=self.max_depth,
                                              random_state=self.random_state, class_weight='balanced')
-            if clf:
+
+            if clf: # FIXME: is 'clf' possibly unbound?
                 clf = clf.fit(x_train, y_train)
                 simple_pred = clf.predict(x_test)
 
         else:
-            raise NotImplementedError(f"expected to be one of ['random', 'constant', 'tree'] \
-                                    but instead got {self.simple_model_type}")
+            raise DeepchecksValueError(
+                f'Unknown model type - {self.simple_model_type}, expected to be one of '
+                f"['random', 'constant', 'tree'] but instead got {self.simple_model_type}" # pylint: disable=inconsistent-quotes
+            )
 
         y_test = test_ds.label_col
 
-        if self.metric is not None:
-            scorer = validate_scorer(self.metric, model, train_ds)
-            metric_name = self.metric_name or self.metric if isinstance(self.metric, str) else 'User metric'
+        if self.scorer is not None:
+            scorer = validate_scorer(self.scorer, model, train_ds)
+            scorer_name = self.scorer if isinstance(self.scorer, str) else 'User Scorer'
         else:
-            metric_name = DEFAULT_SINGLE_METRIC[task_type]
-            scorer = DEFAULT_METRICS_DICT[task_type][metric_name]
+            scorer_name = DEFAULT_SINGLE_METRIC[task_type]
```

Comment:
Also rename these const dicts to DEFAULT_SINGLE_SCORER etc

Reply:
updated